src/pobox.erl

```diff
@@ -40,12 +67,30 @@
 -export_type([max/0, filter/0, in/0, mail/0, note/0]).
 
 -record(state, {buf :: buffer(),
-                owner :: pid(),
+                owner :: name(),
+                owner_pid :: pid(),
                 filter :: undefined | filter(),
-                filter_state :: undefined | term()}).
-
--export([start_link/3, start_link/4, start_link/5, 
-        resize/2, active/3, notify/1, post/2, post_sync/3]).
+                filter_state :: undefined | term(),
+                heir :: pid() | atom(),
+                heir_data :: none | term(),
+                name :: name()}).
+
+-record(pobox_opts, {
+  name = undefined :: undefined | name(),
+  owner = self() :: name(),
+  size = 100 :: non_neg_integer(),
+  type = queue :: stack | queue | keep_old | {mod, module()},
+  state_name = notify :: notify | passive,
+  heir = none :: none | name(),
```

Comment:
I chose to use `name()` here because is the heir dies and is restarted we can always link to whichever one is registered at that moment the owner dies.